processor/lsmintervalprocessor/processor.go

```diff
@@ -65,14 +66,15 @@ const (
 	// a MemTable is being flushed.
 	pebbleMemTableStopWritesThreshold = 2
 
-	// dbCommitThresholdBytes is a soft limit and the batch is committed
-	// to the DB as soon as it crosses this threshold. To make sure that
-	// the commit threshold plays well with the max retained batch size
-	// the threshold should be kept smaller than the sum of max retained
-	// batch size and encoded size of aggregated data to be committed.
-	// However, this requires https://github.com/cockroachdb/pebble/pull/3139.
-	// So, for now we are only tweaking the available options.
-	dbCommitThresholdBytes = 8 << 20 // 8MB
+	// dbCommitDatapointsThresholdCount is the soft limit on the count of
+	// the datapoints after which the value is committed to the database.
+	// The threshold is not based on size in bytes, however, to make sure
+	// that the commit threshold plays well with the max retained batch size
+	// the threshold should be kept smaller than the expected sum of max
+	// retained batch size and encoded size of aggregated data to be
+	// committed. In order to achieve this we can estimate the size of the
+	// datapoint.
+	dbCommitDatapointsThresholdCount = 16384
```

Comment:
[**For reviewers**] I have kept this value as 2x of the batch processor's `send_batch_size` (which was used in the PRs to add batch processor).